src/pkg/services/m365/api/groups.go

```diff
@@ -154,6 +155,75 @@ func (c Groups) GetByID(
 	return group, nil
 }
 
+// GetAllSites gets all the sites that belong to a group. This is
+// necessary as private and shared channels gets their on individual
+// sites. All the other channels make use of the root site.
+func (c Groups) GetAllSites(
+	ctx context.Context,
+	identifier string,
+	errs *fault.Bus,
+) ([]models.Siteable, error) {
+	root, err := c.GetRootSite(ctx, identifier)
+	if err != nil {
+		return nil, clues.Wrap(err, "getting root site")
+	}
+
+	sites := []models.Siteable{root}
+
+	channels, err := Channels(c).GetChannels(ctx, identifier)
+	if err != nil {
+		return nil, clues.Wrap(err, "getting channels")
+	}
+
+	service, err := c.Service()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, ch := range channels {
+		if ptr.Val(ch.GetMembershipType()) == models.STANDARD_CHANNELMEMBERSHIPTYPE {
+			// Standard channels use root site
+			continue
+		}
+
```

Comment:
nit: setup clues first, then reference this ctx throughout the loop.  This has the added benefit of cascading these details to downstream calls as well.
```suggestion
ictx := clues.Add(ctx, "channel_id", ptr.Val(ch.GetId())), "channel_name", clues.Hide(ptr.Val(ch.GetName()))
```